JumpscaleLibs/sal/zosv2/Zosv2.py

```diff
@@ -0,0 +1,197 @@
+from Jumpscale import j
+import netaddr
+import ipaddress
+import socket
+from nacl import public
+from nacl.encoding import Base64Encoder
+class NetworkConfig:
+    def __init__(self, private_key=None,public_key=None,wg_port=None, public_ip=None):
+        self.private_key = private_key
+        self.public_key = public_key
+        self.wg_port = wg_port
+        self.public_ip = public_ip
+
+class Zosv2(j.baseclasses.object):
+    __jslocation__ = "j.sal.zosv2"
+
+    def _init(self, **kwargs):
+        #j.clients.threebot.explorer_addr_set("explorer.testnet.grid.tf")
+        self.explorer = j.clients.threebot.explorer
+        self.config = NetworkConfig()
```

Comment:
This looks very weird. Why would you have a single instance of a network config in the SAL like this ?

Reply:
This is small class to hold all args we need in wire-guard config instead of pass all args in function.